### ISSUE TITLE
Change the use of the term "Weight" to "Priority"

### ISSUE THREAD

**Issue by jjonesvraws** [aws/karpenter-provider-aws#7444]
### Description

In the case of the term Weight being used for a NodePool, this terminology is somewhat misleading and instead should be renamed to Priority.
NodePools are selected based on the weight value and ignores subsequent ones with a lower weight value, when in fact the term weight would suggest the distribution between multiple NodePools.
A weighting would assist where you may want to have a certain percentage of nodes in one NodePool vs another percentage in an alterantive NodePool. For example: 20% On-demand and 80% Spot.

Priority would be a more appropriate term for what is currently known as Weight.

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jonathan-innis** [comment]
> instead should be renamed to Priority

Given that we are GA and this field is part of the stable API, we're not going to be able to change the name of this field in the NodePool. Do you think there's something that we should do in the docs that would make this more clear?

**Comment by jjonesvraws** [comment]
Yes, I think in the opening line here, just to make it absolutely clear that it isnt a weighted behaviour in the sense of a percentage split or proportionate distribution.
https://karpenter.sh/v1.0/concepts/scheduling/#weighted-nodepools

To something along the lines of:

Karpenter allows you to order your NodePools using the .spec.weight field and will opt for the NodePool with the highest weight value first and then subsequently lower weights after, acting as a priority/ordering mechanism for your NodePools. Note: This is not a weighted mechanism to distribute a proportion of nodes across each NodePool
